lib/assert.js

```diff
@@ -151,6 +153,29 @@ x.ifError = function (err, msg) {
 	test(!err, create(err, 'Error', '!==', msg, x.ifError));
 };
 
+x.snapshot = function (tree, optionalMessage, match, snapshotStateGetter) {
+	// set defaults - this allows tests to mock deps easily
+	var toMatchSnapshot = match || jestSnapshot.toMatchSnapshot;
+	var getState = snapshotStateGetter || snapshotState.get;
+
+	var state = getState();
+
+	var context = {
+		dontThrow: function () {},
+		currentTestName: this.title,
+		snapshotState: state
+	};
+
+	var result = toMatchSnapshot.call(context, tree);
+
+	var messagePrefix = 'Please check your code or --update-snapshots\n\n';
+	var message = optionalMessage || (typeof result.message === 'function' && (messagePrefix + result.message()));
```

Comment:
The message argument should be in addition to the default output, not replace it. Probably best to prepend before `messagePrefix`.